src/dos/dos.cpp

```diff
@@ -879,8 +879,11 @@ static Bitu DOS_21Handler(void) {
 				CALLBACK_SCF(true);
 			}
 		} else if (reg_al==0x01) {
-			LOG(LOG_DOSMISC,LOG_ERROR)("DOS:57:Set File Date Time Faked");
-			CALLBACK_SCF(false);		
+			if (DOS_SetFileDate(reg_bx, reg_cx, reg_dx)) {
+				CALLBACK_SCF(false);
+			} else {
+				CALLBACK_SCF(true);
+			}
```

Comment:
Looking at this code, and surrounding area… I think there's a part missing in here, I *suspect* (not sure), the code should look something like this:
```suggestion
			if (DOS_SetFileDate(reg_bx, reg_cx, reg_dx)) {
				reg_ax =  // some value here, I don't know what
				CALLBACK_SCF(false);
			} else {
				reg_ax = dos.errorcode; // this one I am 99% sure
				CALLBACK_SCF(true);
			}
```